src/assisted_installer_controller/assisted_installer_controller.go

```diff
@@ -135,14 +135,21 @@ func (c *controller) WaitAndUpdateNodesStatus(status *ControllerStatus) {
 			if !ok {
 				continue
 			}
-
 			ctx := utils.GenerateRequestContext()
 			log := utils.RequestIDLogger(ctx, c.log)
-			log.Infof("Found new joined node %s with inventory id %s, kubernetes id %s, updating its status to %s",
-				node.Name, host.Host.ID.String(), node.Status.NodeInfo.SystemUUID, models.HostStageDone)
-			if err := c.ic.UpdateHostInstallProgress(ctx, host.Host.ID.String(), models.HostStageDone, ""); err != nil {
-				log.Errorf("Failed to update node %s installation status, %s", node.Name, err)
-				continue
+			if common.IsK8sNodeIsReady(node) {
+				log.Infof("Found new joined node %s with inventory id %s, kubernetes id %s, updating its status to %s",
+					node.Name, host.Host.ID.String(), node.Status.NodeInfo.SystemUUID, models.HostStageDone)
+				if err := c.ic.UpdateHostInstallProgress(ctx, host.Host.ID.String(), models.HostStageDone, ""); err != nil {
```

Comment:
What if the node is already in `Done`?
No reason to spam the service with more updates
Consider adding `models.HostStageDone` to `ignoreStatuses`

Reply:
it will not spam. Node in Done status will not be part of assistedInstallerNodesMap. It will be filtered here : https://github.com/openshift/assisted-installer/pull/206/files#diff-1da579fda3bc689940661d48dd744a46f23824270427207cfae4f8c7448eddd2L134